flink/scripts/jobmanager-prestop-hook.sh

```diff
@@ -25,11 +25,16 @@ cd "$(dirname "${BASH_SOURCE[0]}")"
 source ./common.sh
 
 stop_job() {
-  local cmd_return job_id pattern savepoint_path
-  # TODO get job id from rest api"
+  local cmd_args cmd_return job_id pattern savepoint_path
+  # TODO get job id from rest api
   job_id="00000000000000000000000000000000"
   log_info "Stopping the job with savepoint"
-  cmd_return=$("$flink_home/bin/flink" stop -d "$job_id")
+  cmd_args=("-d" "$job_id")
+  if [ "${FLINK_CONF_JOBMANAGER_RPC_ADDRESS:-undefined}" != "undefined" ]; then
+    cmd_args=("${FLINK_CONF_JOBMANAGER_RPC_ADDRESS}:8081" "${cmd_args[@]}")
```

Comment:
Since this info is already inside flink-conf.yaml, is it really necessary?

by the docs:
```
  Options for default mode:
     -m,--jobmanager <arg>           Address of the JobManager to which
                                     to connect. Use this flag to connect to a
                                     different JobManager than the one specified
                                     in the configuration.
```
If it is necessary for some reason, we could use rest port configured on flink-conf.yaml through env vars:

```suggestion
    cmd_args=("${FLINK_CONF_JOBMANAGER_RPC_ADDRESS}:${FLINK_CONF_REST_PORT}" "${cmd_args[@]}")
```

Reply:
You are right, if flink-conf.yaml is there this option is not necessary. But in helm hooks it isn't. I has two options here:
1 - create flink conf before call pre-stop hook script. (More steps to fail)
2 - do not assume flink-conf is there and pass in the jobmanager address (error prone if environment variables changes during execution)

I chose 2. What do you think?

Reply:
Got it. 2 :)